### PR TITLE
Feature/add silent flag

### PR DESCRIPTION
**Pull request by bronzdoc**
Adding  install/update --silent flag to silence output
closes https://github.com/rubygems/rubygems/issues/787

### CODE REVIEW

File: test/rubygems/test_gem_commands_install_command.rb [hunk 1]
```diff
@@ -95,6 +95,34 @@ def test_execute_local
...
+    end
+
+    @cmd.options[:domain] = :local
+
+    FileUtils.mv specs['a-2'].cache_file, @tempdir
+
+    @cmd.handle_options %w[a --silent]
+
+    use_ui @ui do
+      orig_dir = Dir.pwd
```
Comment (segiddins) [inline test/rubygems/test_gem_commands_install_command.rb]: `Dir.chdir(@tmpdir) do` ?

**Comment by segiddins** [comment]
:+1: other than the one comment

File: test/rubygems/test_gem_commands_install_command.rb (hunk 1 above)
Comment (bronzdoc) [inline test/rubygems/test_gem_commands_install_command.rb]: Thanks, i update it

Comment (segiddins) [inline test/rubygems/test_gem_commands_install_command.rb]: the variable is now unused

**Comment by drbrain** [comment]
This should be a [common option](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/command.rb#L512-L524), not specific to particular commands.

**Comment by bronzdoc** [comment]
Ok @drbrain will update it

File: test/rubygems/test_gem_commands_install_command.rb (hunk 1 above)
Comment (bronzdoc) [inline test/rubygems/test_gem_commands_install_command.rb]: damn it lol

**Comment by bronzdoc** [comment]
@drbrain i updated it
\cc @segiddins

**Comment by indirect** [comment]
Thanks!

**Pull request merged**